src/Service/VendorService/EReolenGlobal/EReolenGlobalVendorService.php

```diff
@@ -0,0 +1,207 @@
+                $this->progressAdvance();
+
+                $offset += self::BATCH_SIZE;
+            } while ($offset < $totalCount);
+
+            $this->logStatistics();
+
+            $this->progressFinish();
+
+            return VendorImportResultMessage::success($this->totalIsIdentifiers, $this->totalUpdated, $this->totalInserted, $this->totalDeleted);
+        } catch (\Exception $exception) {
+            return VendorImportResultMessage::error($exception->getMessage());
+        }
+    }
+
+    /**
+     * Get products from api.
+     *
+     * @param string $productsUrl
+     * @param int $limit
+     * @param int $offset
+     *
+     * @return array
+     *
+     * @throws IllegalVendorServiceException
+     * @throws UnknownVendorServiceException
+     * @throws GuzzleException
+     */
+    private function getProducts(string $productsUrl, int $limit, int $offset): array
+    {
+        $response = $this->httpClient->request('GET', $productsUrl, [
+            'headers' => [
+                'User-Agent' => $this->getVendor()->getDataServerUser(),
+                'Authorization' => 'Bearer '.$this->getAuthToken(),
+            ],
+            'query' => ['limit' => $limit, 'offset' => $offset],
+        ]);
+
+        $content = $response->getBody()->getContents();
+        $content = json_decode($content);
+
+        return $content->products;
+    }
+
+    /**
+     * Get the total number of products from api.
+     *
+     * @param string $productsUrl
+     *
+     * @return int
+     *
+     * @throws GuzzleException
+     * @throws IllegalVendorServiceException
+     * @throws UnknownVendorServiceException
+     */
+    private function getTotalItems(string $productsUrl): int
+    {
+        $response = $this->httpClient->request('GET', $productsUrl, [
+            'headers' => [
+                'User-Agent' => $this->getVendor()->getDataServerUser(),
+                'Authorization' => 'Bearer '.$this->getAuthToken(),
+            ],
+        ]);
+
+        $content = $response->getBody()->getContents();
+        $content = json_decode($content);
+
+        return $content->totalItems;
+    }
+
+    /**
+     * Get Overdrive auth token.
+     *
+     * @see https://developer.overdrive.com/apis/client-auth
+     *
+     * @return string
+     *
+     * @throws GuzzleException
+     * @throws IllegalVendorServiceException
+     * @throws UnknownVendorServiceException
+     */
+    private function getAuthToken(): string
+    {
+        $response = $this->httpClient->request('POST', self::AUTH_URL, [
+                'auth' => [$this->getVendor()->getDataServerUser(), $this->getVendor()->getDataServerPassword()],
+                'headers' => ['Content-Type' => 'application/x-www-form-urlencoded;charset=UTF-8'],
+                'body' => self::AUTH_GRANT_TYPE,
+        ]);
+
+        $content = $response->getBody()->getContents();
+        $content = json_decode($content);
+
+        return $content->access_token;
```

Comment:
Should the token be cached or is `getAuthToken` only called once during the importers run. oAuth tokens normally comes with an expire timestamp.

Reply:
It is only called once pr. import run. TTL is one hour. So in production caching will be redundant because we don't run the imports that often